homeassistant/components/keba/binary_sensor.py

```diff
@@ -0,0 +1,147 @@
+"""Support for KEBA charging station binary sensors."""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSORS = {
+    'Online': ['Wallbox', 'connectifity'],
+    'Plug': ['Plug', 'plug'],
+    'State': ['Charging state', 'power'],
+    'Tmo FS': ['Failsafe Mode', 'safety'],
+}
+
+
+async def async_setup_platform(hass, config,
+                               async_add_entities, discovery_info=None):
+    """Set up the KEBA charging station platform."""
+    _LOGGER.debug("Initializing KEBA charging station binary sensors")
+    sensors = []
+    keba = hass.data[DOMAIN]
+
+    for key, value in sorted(SENSORS.items()):
+        sensors.append(KebaBinarrySensor(key, keba, value[0], value[1]))
+
+    async_add_entities(sensors)
+
+
+class KebaBinarrySensor(BinarySensorDevice):
+    """Representation of a binary sensor of a KEBA charging station."""
+
+    def __init__(self, key, keba, sensor_name, device_class):
+        """Initialize the KEBA Sensor."""
+        self._key = key
+        self._keba = keba
+        self._name = sensor_name
+        self._device_class = device_class
+        self._state = None
+        self._attributes = {}
+
+    @property
+    def should_poll(self):
+        """Return False.
+
+        Data update is triggered from Keba entitiy.
+        """
+        return False
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the binary sensor."""
+        return "keba_" + self._name
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return "keba_" + self._name
```

Comment:
This is duplicated for unique id, how about returning the `unique_id` here (or other way around)?

Reply:
Name now only includes the name of the sensor.